iib/workers/tasks/build.py

```diff
@@ -1431,3 +1469,43 @@ def _add_ocp_label_to_index(ocp_version, temp_dir, dockerfile_name):
         log.debug(
             'Added the following line to %s: %s', dockerfile_name, label,
         )
+
+
+def _add_distribution_scope_label_to_index(distribution_scope, temp_dir, dockerfile_name):
+    """
+    Add the distribution scope to the provided dockerfile.
+
+    :param str distribution_scope: the distribution_scope that the index is for.
+    :param str temp_dir: the temp directory to look for the dockerfile.
+    :param str dockerfile_name: the dockerfile name.
+    """
+    with open(os.path.join(temp_dir, dockerfile_name), 'a') as dockerfile:
+        label = f'LABEL com.redhat.index.delivery.distribution_scope="{distribution_scope}"'
+        dockerfile.write(f'\n{label}\n')
+        log.debug(
+            'Added the following line to %s: %s', dockerfile_name, label,
+        )
+
+
+def _validate_distribution_scope(resolved_distribution_scope, distribution_scope):
+    """
+    Validate distribution scope is allowed to be updated.
+
+    :param str resolved_distribution_scope: the distribution_scope that the index is for.
+    :param str distribution_scope: the distribution_scope that has been requested the index is for.
+    :return: the valid distribution_scope
+    :rtype: str
+    :raises IIBError: if the resolved_distribution_scope is of lesser scope then distribution_scope
+    """
+    resolved_distribution_scope = resolved_distribution_scope.upper()
+    distribution_scope = (distribution_scope or resolved_distribution_scope).upper()
+
+    if (resolved_distribution_scope == 'STAGE' and distribution_scope == 'PROD') or (
+        resolved_distribution_scope == 'DEV'
+        and (distribution_scope == 'STAGE' or distribution_scope == 'PROD')
+    ):
```

Comment:
This is hard to read and it'll make the if statement even more complicated if you add a distribution scope later on. How about this?
```suggestion
    distribution_scopes = ["DEV", "STAGE", "PROD"]
    # Make sure the request isn't regressing the distribution scope
    if distribution_scopes.index(distribution_scope) < distribution_scopes.index(resolved_distrbution_scope):
```